apps/07jquery-tic-tac-toe/script.js

```diff
@@ -1,50 +1,82 @@
+// Coded by Alex Gaw. If you judge me for the lightEmUp function, I'll judge you for judging me. Fair warning!
+
 'use strict';
 
 $(document).on('ready', function() {
   // Put app logic in here
   var playerTurn = 'X';
   var play = true;
+  var turns = 0;
+  var winCombinations = [
+    [0, 1, 2], [3, 4, 5], [6, 7, 8],
+    [0, 3, 6], [1, 4, 7], [2, 5, 8],
+    [0, 4, 8], [2, 4, 6]
+  ];
+
+  // Set up #announce-winner with text so it takes up space. Visibility is set to hidden until someone wins.
+  $('#announce-winner').html('Player <span id="winner">X</span> Wins!');
 
   function stopGame() {
     $('div[data-cell]:empty').html('<span></span>');
     play = false;
   }
 
+  function getText(cell) {
+    return $('div[data-cell="' + cell + '"]').text();
+  }
+
+
+
   function checkForWin() {
-    // Check for horizontal win
-    if (($("div[data-cell='0']").text() === playerTurn && $("div[data-cell='1']").text() === playerTurn && $("div[data-cell='2']").text() === playerTurn) || ($("div[data-cell='3']").text() === playerTurn && $("div[data-cell='4']").text() === playerTurn && $("div[data-cell='5']").text() === playerTurn) || ($("div[data-cell='6']").text() === playerTurn && $("div[data-cell='7']").text() === playerTurn && $("div[data-cell='8']").text() === playerTurn)) {
-      stopGame();
-      return true;
-    }
-    // Check for vertical win
-    if (($("div[data-cell='0']").text() === playerTurn && $("div[data-cell='3']").text() === playerTurn && $("div[data-cell='6']").text() === playerTurn) || ($("div[data-cell='1']").text() === playerTurn && $("div[data-cell='4']").text() === playerTurn && $("div[data-cell='7']").text() === playerTurn) || ($("div[data-cell='2']").text() === playerTurn && $("div[data-cell='5']").text() === playerTurn && $("div[data-cell='8']").text() === playerTurn)) {
-      stopGame();
-      return true;
+    var winArray = [];
+    for (var i = 0; i < 8; i++) {
+      for (var j = 0; j < 3; j++) {
+        if (getText(winCombinations[i][j]) === playerTurn) {
+          winArray.push($('div[data-cell="' + winCombinations[i][j] + '"]'));
```

Comment:
Thanks, but this isn't the most recent version. The more recent version is shorter and more functional.